src/dvclive/lightning.py

```diff
@@ -81,12 +81,16 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None):
         ), "experiment tried to log from global_rank != 0"
 
         self.experiment.step = step
+        epoch = False
         for metric_name, metric_val in metrics.items():
             if is_tensor(metric_val):
                 metric_val = metric_val.cpu().detach().item()
+            if metric_name.endswith("_epoch"):
+                epoch = True
             metric_name = standardize_metric_name(metric_name, __name__)
             self.experiment.log_metric(name=metric_name, val=metric_val)
-        self.experiment.next_step()
+        if epoch:
+            self.experiment.next_step()
```

Comment:
Because we already set `step` above:

```suggestion
            self.experiment.make_summary()

            if self.experiment._dvcyaml:
                self.experiment.make_dvcyaml()

            self.experiment.make_report()
```